db/diff.go

```diff
@@ -23,17 +23,17 @@ func fullSync(db *DB, from hash.Hash) ([]kv.Operation, Commit) {
 	return r, makeCommit(db.Noms(), types.Ref{}, datetime.Epoch, db.ds.Database().WriteValue(m.NomsMap()), m.NomsChecksum(), 0 /*lastMutationID*/)
 }
 
-func (db *DB) Diff(from hash.Hash, fromChecksum kv.Checksum) ([]kv.Operation, error) {
+func (db *DB) Diff(fromHash hash.Hash, fromChecksum kv.Checksum, to Commit) ([]kv.Operation, error) {
```

Comment:
I know it's not part of this change, but why does this function require `fromChecksum`? It already has a Noms hash (presumably because client sent it) -- could the checksum really not match?

Reply:
AIUI the protocol treats the state id as an opaque token and it is just convenient (and helpful) that it is a noms hash. The mechansim we have to ensure the two maps are identical is the checksum, so seems prudent to use it for that purpose. I admit that I have a hard time imagining how they might not match, but you don't build safety nets if you can see all the potential problems I guess.